projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js

```diff
@@ -14,43 +19,87 @@ function requiresPlan( unavailableReason, details ) {
 /**
  * Registers a gutenberg block if the availability requirements are met.
  *
- * @param {string} name - The block's name.
+ * @param {string} nameOrMetadata - The block's name or metadata object. Jetpack blocks must be
+ * registered with a name prefixed with `jetpack/`. This function accepts an unprefixed name too,
+ * though (it'd handle both `business-hours` and `jetpack/business-hours` similarly, for instance).
  * @param {object} settings - The block's settings.
  * @param {object} childBlocks - The block's child blocks.
+ * @param {boolean} prefix - Should this block be prefixed with `jetpack/`?
  * @returns {object|boolean} Either false if the block is not available, or the results of `registerBlockType`
  */
-export default function registerJetpackBlock( name, settings, childBlocks = [] ) {
-	const { available, details, unavailableReason } = getJetpackExtensionAvailability( name );
+export default function registerJetpackBlock(
+	nameOrMetadata,
+	settings,
+	childBlocks = [],
+	prefix = true
+) {
+	const name = typeof nameOrMetadata === 'string' ? nameOrMetadata : nameOrMetadata.name;
+	const isNamePrefixed = name.startsWith( JETPACK_PREFIX );
+	const rawName = isNamePrefixed ? name.slice( JETPACK_PREFIX.length ) : name;
+
+	const { available, details, unavailableReason } = getJetpackExtensionAvailability( rawName );
 
 	const requiredPlan = requiresPlan( unavailableReason, details );
```

Comment:
**Discrepancy between native and web version:** In the web version, we are using `requiresPaidPlan` instead of `requiresPlan`.

https://github.com/Automattic/jetpack/blob/eebad7d58da476499ccc090d3b78c02547918c33/projects/plugins/jetpack/extensions/shared/register-jetpack-block.js#L35

AFAIK we are not filtering yet blocks in the native version depending on the plan, so probably this it not an actual issue now.

Reply:
That's a good spot, I see that there's a related `requiresPlan` function in the native file, but don't believe it's currently doing much:

https://github.com/Automattic/jetpack/blob/eebad7d58da476499ccc090d3b78c02547918c33/projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js#L12-L18

As we're not changing either of these lines with this PR, I'm hesitant to tweak them, as I don't want to inadvertently expand the scope of this fix. I've opened a separate GitHub issue, though, to track the issue of filtering blocks based on plan: https://github.com/wordpress-mobile/gutenberg-mobile/issues/6353

Let me know if you think that addresses the problem you've raised here!

Reply:
> As we're not changing either of these lines with this PR, I'm hesitant to tweak them, as I don't want to inadvertently expand the scope of this fix. I've opened a separate GitHub issue, though, to track the issue of filtering blocks based on plan: https://github.com/wordpress-mobile/gutenberg-mobile/issues/6353
> 
> Let me know if you think that addresses the problem you've raised here!

Oh, I didn't mean to imply that we should fix it here, I just wanted to share the discrepancies I found when comparing both versions in case there was something we'd need to follow up with. Tracking this in a GitHub issue sounds good, thanks 🙇 !